superset/assets/src/explore/components/ExploreChartHeader.jsx

```diff
@@ -135,12 +134,12 @@ class ExploreChartHeader extends React.PureComponent {
           />
         }
         <div className="pull-right">
-          {chartSucceeded && queryResponse &&
+          {queryResponse &&
             <RowCountLabel
               rowcount={queryResponse.rowcount}
               limit={formData.row_limit}
             />}
-          {chartSucceeded && queryResponse && queryResponse.is_cached &&
+          {queryResponse && queryResponse.is_cached &&
```

Comment:
The logic here says we should only render cache label when
- fetch data succeeded, and
- has query response, and
- queryResponse.is_cached is true.

in No data case, we should not render cache label, and should not render a label and ask user to force refresh.